### PR TITLE
TypeScript

### PR DESCRIPTION
**Pull request by markdstouffer**
# Types!
- Use included typings from Discord.js library
- Create/enforce type interfaces for mongoose docs & AniList API data

## Some other changes
- Improved error handling throughout commands
- Fix `/ep next` issue - now checks for existing current WP before allowing users to progress episodes

### CODE REVIEW

**Comment by markdstouffer** [comment]
@cakoshakib ready for review. checks that didn't pass are defunct anyways now, WHO LINTS MARKDOWN... awaiting approval.

**Comment by cakoshakib** [comment]
Consider using the markdown linter extension on VSCode, noob.

File: src/commands/remind-quick.ts [hunk 1]
```diff
@@ -1,26 +1,30 @@
...
 
 module.exports = {
   data: {
     name: 'remind-quick',
-    type: 2,
+    type: 2
   },
-  async execute(interaction) {
+  async execute (interaction: CommandInteraction) {
     const discord = interaction.options.getUser('user')
```
Comment (cakoshakib) [inline src/commands/remind-quick.ts:14]: might wanna add a type here

File: src/commands/url-quick.ts [hunk 2]
```diff
@@ -9,20 +13,20 @@ module.exports = {
     name: 'url-quick',
     type: 2
   },
-  async execute(interaction) {
+  async execute (interaction: CommandInteraction) {
     const discord = interaction.options.getUser('user')
```
Comment (cakoshakib) [inline src/commands/url-quick.ts:17]: probably here too

File: src/commands/remind.ts [hunk 3]
```diff
@@ -6,30 +11,30 @@ module.exports = {
...
         .setName('discord')
         .setDescription('Discord tag')
         .setRequired(true)
-      ),
-  async execute(interaction) {
+    ),
+  async execute (interaction: CommandInteraction) {
     const serverId = interaction.guildId
     const query = { 'server.serverId': serverId }
     const discord = interaction.options.getMentionable('discord')
```
Comment (cakoshakib) [inline src/commands/remind.ts:23]: prob here too

**Review by cakoshakib — COMMENTED** [review]
Looks good! I don't know TS so these might mean nothing, but I think types in these locations make sense.

**Pull request merged**